app/src/main/java/org/y20k/transistor/core/Station.java

```diff
@@ -293,6 +303,22 @@ private ContentType getContentType(URL fileLocation) {
                     if (charsetString != null) {
                         contentType.charset = charsetString.trim();
                     }
+                    // Generic octet-stream MIME type, check if we can guess file type from extension
```

Comment:
Perhaps we should do this check also if no content type was provided in the response head, i.e. `contentTypeHeader == null`.